### PR TITLE
RHOAIENG-3783 - Fix codeflare sdk tests

### PR DESCRIPTION
**Pull request by Srihari1192**
closes [RHOAIENG-3783](https://issues.redhat.com/browse/RHOAIENG-3783) and [RHOAIENG-4754](https://issues.redhat.com/browse/RHOAIENG-4754)
- Added scripts for python 3.9 Virtual environment setup for codeflare-sdk python tests
-  Updated the Go tests to CodeFlare SDK python tests
- Add the SDK authentication test
- Added timeout to maximum 10 minutes and tests running successfully less than 5 min

### CODE REVIEW

**Comment by Srihari1192** [comment]
@jiripetrlik @sutaakar While testing in CI noticed below error in tear down at  [step](https://github.com/red-hat-data-services/ods-ci/pull/1275/files#diff-c4f1da422d82b1d191c5a90d8549babd63160c412f05b344e4daa5ea51749b30R17-R19) 

```
13:12:12  Component codeflare state was set to Removed
13:12:12  Component ray state was set to Removed
13:12:12  ./ods_ci/tests/Resources/Page/DistributedWorkloads/cleanup_python_virtual_env.sh: line 19: sudo: command not found
13:12:12  Virtual environment venv3.9 deleted successfully.
```

Is there any other approach i can use here , without sudo permission denied error throwing

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:68]: IMHO it would have more sense to install latest released SDK version.
Current implementation test latest main branch, which won't be used by users. Users will use latest release (or version which is bundled in Notebook images).

**Comment by sutaakar** [comment]
> Is there any other approach i can use here , without sudo permission denied error throwing

No idea

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot
Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:68]: Yes. May be fetch the codeflare sdk version from[ notebooks pip file ](https://github.com/opendatahub-io/notebooks/blob/cdc08c6e846c02404fa119cab5cfe71eecba5ebf/jupyter/datascience/ubi9-python-3.9/Pipfile#L19) , clone and test against that released tag would make sense

Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:77]: Done

**Comment by Srihari1192** [comment]
> @jiripetrlik @sutaakar While testing in CI noticed below error in tear down at [step](https://github.com/red-hat-data-services/ods-ci/pull/1275/files#diff-c4f1da422d82b1d191c5a90d8549babd63160c412f05b344e4daa5ea51749b30R17-R19)
> 
> ```
> 13:12:12  Component codeflare state was set to Removed
> 13:12:12  Component ray state was set to Removed
> 13:12:12  ./ods_ci/tests/Resources/Page/DistributedWorkloads/cleanup_python_virtual_env.sh: line 19: sudo: command not found
> 13:12:12  Virtual environment venv3.9 deleted successfully.
> ```
> 
> Is there any other approach i can use here , without sudo permission denied error throwing

still need to figure it out for uninstall python error in CI.. Will it make sense to remove only python virtual environment?

**Review by jiripetrlik — APPROVED** [review]

**Review by sutaakar — DISMISSED** [review]

**Comment by Srihari1192** [comment]
Tests were working fine against the main branch after this changes [TokenAuthentication](https://github.com/project-codeflare/codeflare-sdk/pull/487) in **codeflare-sdk** oauth tests.. Since we are testing with latest tag , i think we need to wait until latest codeflare-sdk new release to merge this PR

**Comment by Srihari1192** [comment]
> [Tests](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/distributed-workloads/job/rhods-pipeline-dw-new/13/) were working fine against the main branch after this changes [TokenAuthentication](https://github.com/project-codeflare/codeflare-sdk/pull/487) in **codeflare-sdk** oauth tests.. Since we are testing with latest tag , i think we need to wait until latest codeflare-sdk new release to merge this PR

@jiripetrlik @sutaakar Modify the tests to run against main branch and created a [RHOAIENG-5070](https://issues.redhat.com/browse/RHOAIENG-5070) to track to update to run tests against latest tag

**Review by jiripetrlik — APPROVED** [review]

**Review by ChughShilpa — APPROVED** [review]

**Pull request merged**
